mobgap/gsd/evaluation.py

```diff
@@ -402,59 +417,165 @@ def _get_false_matches_from_overlap_data(overlaps: list[Interval], interval: Int
-    start and end indices of the respective gait sequences.
+    # add tn intervals
+    tn_intervals = []
+    for i, (start, _) in enumerate(categorized_intervals[["start", "end"]].itertuples(index=False)):
+        if i == 0:
+            if start > 0:
+                tn_intervals.append([0, start])
+        elif start > categorized_intervals.iloc[i - 1]["end"]:
+            tn_intervals.append([categorized_intervals.iloc[i - 1]["end"], start])
+
+    if categorized_intervals.iloc[-1]["end"] < n_overall_samples - 1:
+        tn_intervals.append([categorized_intervals.iloc[-1]["end"], n_overall_samples - 1])
+
+    tn_intervals = pd.DataFrame(tn_intervals, columns=["start", "end"])
+    tn_intervals["match_type"] = "tn"
+    return tn_intervals
+
+
+def plot_categorized_intervals(
+    gsd_list_detected: pd.DataFrame, gsd_list_reference: pd.DataFrame, categorized_intervals: pd.DataFrame
+) -> Figure:
+    """Plot the categorized intervals together with the detected and reference intervals."""
+    fig, ax = plt.subplots(figsize=(10, 3))
+    _plot_intervals_from_df(gsd_list_reference, 3, ax, color="orange")
+    _plot_intervals_from_df(gsd_list_detected, 2, ax, color="blue")
+    _plot_intervals_from_df(categorized_intervals.query("match_type == 'tp'"), 1, ax, color="green", label="TP")
+    _plot_intervals_from_df(categorized_intervals.query("match_type == 'fp'"), 1, ax, color="red", label="FP")
+    _plot_intervals_from_df(categorized_intervals.query("match_type == 'fn'"), 1, ax, color="purple", label="FN")
+    plt.yticks([1, 2, 3], ["Categorized", "Detected", "Reference"])
+    plt.ylim(0, 4)
+    plt.xlabel("Index")
+    leg = plt.legend(loc="upper right", bbox_to_anchor=(1, 1.2), ncol=3, frameon=False)
+    for handle in leg.legend_handles:
+        handle.set_linewidth(10)
+    plt.tight_layout()
+    return fig
+
+
+def _sanitize_index(ic_list: pd.DataFrame, list_type: Literal["detected", "reference"]) -> tuple[pd.DataFrame, bool]:
+    is_multindex = False
+    # check if index is a multiindex and raise warning if it is
+    if isinstance(ic_list.index, pd.MultiIndex):
+        is_multindex = True
+        ic_list.index = ic_list.index.to_flat_index()
+        ic_list.index.name = f"gs_id_{list_type}"
+    # check if indices are unique
+    if not ic_list.index.is_unique:
+        raise ValueError(f"The index of `gs_list_{list_type}` must be unique!")
+    return ic_list, is_multindex
+
+
+def _plot_intervals_from_df(df: pd.DataFrame, y: int, ax: Axes, **kwargs: Unpack[dict[str, Any]]) -> None:
+    label_set = False
+    for _, row in df.iterrows():
+        label = kwargs.pop("label", None)
+        if label and not label_set:
+            ax.hlines(y, row["start"], row["end"], lw=20, label=label, **kwargs)
+            label_set = True
+        else:
+            ax.hlines(y, row["start"], row["end"], lw=20, **kwargs)
+
+
+def categorize_matches_with_min_overlap(
```

Comment:
Does categorize "matches" make sense from a naming perspective? Aren't we categorizing intervals? Or do you think this woul dbe confusing with the other `categorize_interval` function?

Reply:
Maybe `categorize_intervals` --> `categorize_intervals_sample_wise` and `categorize_matches_with_min_overlap` --> `categorize_intervals_interval_wise` to make clear whats the difference?